### ISSUE TITLE
https://shields.io website: Access-Control-Allow-Origin header has wrong value

### ISSUE THREAD

**Issue by mprins** [badges/shields#779]
When I go to https://shields.io (and because I have "https everywhere" enabled http://shields.io as well) and enter a url to suggest badges Chrome Version 52.0.2743.116 (64-bit) will fail the request with the following error in the javascript console:

```
XMLHttpRequest cannot load https://img.shields.io/$suggest/v1?url=https%3A%2F%2Fgithub.com%2Fmprins%2Fmaven-compass-example. The 'Access-Control-Allow-Origin' header has a value 'http://shields.io' that is not equal to the supplied origin. Origin 'https://shields.io' is therefore not allowed access.
```
### General

```
Request URL:https://img.shields.io/$suggest/v1?url=https%3A%2F%2Fgithub.com%2Fmprins%2Fmaven-compass-example
Request Method:GET
Status Code:200 
Remote Address:104.18.44.160:443
```
### Response Headers

```
access-control-allow-origin:http://shields.io
cf-ray:2da6e37d4ccd0c7d-AMS
content-encoding:gzip
content-type:application/json
date:Tue, 30 Aug 2016 08:31:50 GMT
server:cloudflare-nginx
set-cookie:__cfduid=d0ae334ae93cbe753d7bcc044762f4d721472545909; expires=Wed, 30-Aug-17 08:31:49 GMT; path=/; domain=.shields.io; HttpOnly
status:200
```
### Request Headers

```
:authority:img.shields.io
:method:GET
:path:/$suggest/v1?url=https%3A%2F%2Fgithub.com%2Fmprins%2Fmaven-compass-example
:scheme:https
accept:*/*
accept-encoding:gzip, deflate, sdch, br
accept-language:nl,en-GB;q=0.8,en;q=0.6,de-AT;q=0.4,de;q=0.2
cache-control:no-cache
dnt:1
origin:https://shields.io
pragma:no-cache
user-agent:Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36
```
### Query String Parameters

```
url:https://github.com/mprins/maven-compass-example
```

It seems the value for `Access-Control-Allow-Origin` is [hardcoded](https://github.com/badges/shields/blob/master/suggest.js#L17) instead of returning the value provided in the `origin` request header, as recommended by the [W3C](https://www.w3.org/TR/cors/#access-control-allow-origin-response-header)

**Comment by leizongmin** [comment]
met the same problem

**Comment by paulmelnikow** [comment]
Thanks for this report. It appears to have been fixed in 9816a50. I can no longer reproduce the problem at https://shields.io/.
